metadata-utils/src/main/java/com/linkedin/metadata/utils/metrics/MetricUtils.java

```diff
@@ -1,51 +1,133 @@
     final JmxReporter reporter = JmxReporter.forRegistry(REGISTRY).build();
     reporter.start();
   }
 
-  public static MetricRegistry get() {
+  public static MetricRegistry getCodaHaleRegistry() {
     return REGISTRY;
   }
 
-  public static Counter counter(Class<?> klass, String metricName) {
-    return REGISTRY.counter(MetricRegistry.name(klass, metricName));
+  public static PrometheusMeterRegistry getMicrometerRegistry() {
+    return PROMETHEUS_METER_REGISTRY;
   }
 
-  public static void exceptionCounter(Class<?> klass, String metricName, Throwable t) {
-    String[] splitClassName = t.getClass().getName().split("[.]");
-    String snakeCase = splitClassName[splitClassName.length - 1].replaceAll("([A-Z][a-z])", DELIMITER + "$1");
+  public static String buildName(String name, String... names) {
+    return MetricRegistry.name(name, names);
+  }
+
+  public static String scrapePrometheusData() {
+    return PROMETHEUS_METER_REGISTRY.scrape();
+  }
+
+  // Counter
+
+  public static void counterInc(long value, String name, String...names) {
+    String finalName = buildName(name, names);
+    REGISTRY.counter(finalName).inc(value);
+    PROMETHEUS_METER_REGISTRY.counter(finalName).increment(value);
+  }
+
+  public static void counterInc(String name, String...names) {
+    String finalName = buildName(name, names);
+    REGISTRY.counter(finalName).inc();
+    PROMETHEUS_METER_REGISTRY.counter(finalName).increment();
+  }
+
+  // Histogram
 
-    counter(klass, metricName).inc();
-    counter(klass, metricName + DELIMITER + snakeCase).inc();
+  public static void updateHistogram(long value, String name, String...names) {
+    String finalName = buildName(name, names);
+    REGISTRY.histogram(finalName).update(value);
+    histogram(finalName).record(value);
   }
 
-  public static Counter counter(String metricName) {
-    return REGISTRY.counter(MetricRegistry.name(metricName));
+  private static DistributionSummary histogram(String name) {
+    if (!HISTOGRAM_MAP.containsKey(name)) {
+      HISTOGRAM_MAP.put(name, DistributionSummary.builder(name)
+              .publishPercentiles(0.75, 0.95, 0.98, 0.99, 0.999)
+              .register(PROMETHEUS_METER_REGISTRY));
+    }
+    return HISTOGRAM_MAP.get(name);
   }
 
-  public static Timer timer(Class<?> klass, String metricName) {
-    return REGISTRY.timer(MetricRegistry.name(klass, metricName));
+  public static DistributionSummary getMicrometerHistogram(String name) {
+    return HISTOGRAM_MAP.get(name);
   }
 
-  public static Timer timer(String metricName) {
-    return REGISTRY.timer(MetricRegistry.name(metricName));
+  // Timer
+
+  public static UUID timerStart(String name, String...names) {
+    String finalName = buildName(name, names);
+    UUID uuid = UUID.randomUUID();
+    TIMER_CONTEXT.put(uuid, REGISTRY.timer(finalName).time());
+    SMAPLE_MAP.put(uuid, Timer.start());
+    return uuid;
+  }
+
+  public static void timerStop(UUID uuid, String name, String...names) {
```

Comment:
if you need the UUID to find the exact metric, then the name is not unique identifier of the metric, which will cause name collision  when we scrape it. 
so UUID is not needed, or we already have a problem.